Neos.Flow/Classes/Http/BaseRequest.php

```diff
@@ -0,0 +1,389 @@
+    {
+        if ($this->baseUri === null) {
+            $this->detectBaseUri();
+        }
+
+        return $this->baseUri;
+    }
+
+    /**
+     * @param Uri $baseUri
+     */
+    public function setBaseUri($baseUri)
+    {
+        $this->baseUri = $baseUri;
+    }
+
+    /**
+     * Tries to detect the base URI of request.
+     *
+     * @return void
+     */
+    protected function detectBaseUri()
+    {
+        if ($this->baseUri === null) {
+            $this->baseUri = clone $this->uri;
+            $this->baseUri->setQuery(null);
+            $this->baseUri->setFragment(null);
+            $this->baseUri->setPath($this->getScriptRequestPath());
+        }
+    }
+
+    /**
+     * Sets the request method
+     *
+     * @param string $method The request method, for example "GET".
+     * @return void
+     * @api
+     */
+    public function setMethod($method)
+    {
+        $this->method = strtoupper($method);
+    }
+
+    /**
+     * Returns the request method
+     *
+     * @return string The request method
+     * @api
+     */
+    public function getMethod()
+    {
+        return $this->method;
+    }
+
+    /**
+     * Explicitly sets the content of the request body
+     *
+     * In most cases, content is just a string representation of the request body.
+     * In order to reduce memory consumption for uploads and other big data, it is
+     * also possible to pass a stream resource. The easies way to convert a local file
+     * into a stream resource is probably: $resource = fopen('file://path/to/file', 'rb');
+     *
+     * @param string|resource $content The body content, for example arguments of a PUT request, or a stream resource
+     * @return void
+     * @api
+     */
+    public function setContent($content)
+    {
+        if (is_resource($content) && get_resource_type($content) === 'stream' && stream_is_local($content)) {
+            $streamMetaData = stream_get_meta_data($content);
+            $this->headers->set('Content-Length', filesize($streamMetaData['uri']));
+            $this->headers->set('Content-Type', MediaTypes::getMediaTypeFromFilename($streamMetaData['uri']));
+        }
+
+        parent::setContent($content);
+    }
+
+    /**
+     * Returns the content of the request body
+     *
+     * If the request body has not been set with setContent() previously, this method
+     * will try to retrieve it from the input stream. If $asResource was set to TRUE,
+     * the stream resource will be returned instead of a string.
+     *
+     * If the content which has been set by setContent() originally was a stream
+     * resource, that resource will be returned, no matter if $asResource is set.
+     *
+     *
+     * @param boolean $asResource If set, the content is returned as a resource pointing to PHP's input stream
+     * @return string|resource
+     * @api
+     * @throws Exception
+     */
+    public function getContent($asResource = false)
+    {
```

Comment:
does it have to be this way? I think that returning two different types depending on a given argument is a bad habit. I'd rather have `getContentAsResource` and `getContent` / `getContentAsString`.